packages/fs.go

```diff
@@ -73,8 +80,8 @@ func NewZipPackageFileSystem(p *Package) (*ZipPackageFileSystem, error) {
 	var root string
 	found := false
 	for _, f := range reader.File {
-		name := filepath.Clean(f.Name)
-		parts := strings.Split(name, string(filepath.Separator))
+		name := path.Clean(f.Name)
+		parts := strings.Split(name, "/")
 		if len(parts) == 2 && parts[1] == "manifest.yml" {
```

Comment:
Nit. Maybe we can use path functions here instead of manipulating the path as a string.
```suggestion
		dir, base := path.Split(name)
		if path.Dir(dir) == "." && base == "manifest.yml" {
```

Reply:
@jsoriano Seems the suggested nit doesn't work as is. I'm not exactly sure what the original intent is, but in the tests the condition
```go
if len(parts) == 2 && parts[1] == "manifest.yml"
```
is only true when the given path is `"example-1.0.1/manifest.yml"`. In this case, `path.Dir(dir)` is equal to `"example-1.0.1"`.

The other case where the file name is equal to `"manifest.yml"` is when the given path is `"example-1.0.1/data_stream/foo/manifest.yml"`, in which case `path.Dir(dir)` is equal to `"example-1.0.1/data_stream/foo"`.

There are no cases where `path.Dir(dir)` is equal to `"."` and the file name is equal to `"manifest.yml"`.

Please let me know if I'm missing something.

Reply:
Thank you for the explanations and playground! I would be inclined to add comment to make the intent explicit, thereby clarifying the partial condition `dir != "." && path.Dir(dir) == "."`. Knowing this, I prefer this option.